### PR TITLE
Deepen calendar time-window handling

### PR DESCRIPTION
**Pull request by Pan14ek**
## Purpose

Deepen calendar event scheduling internals so create, update, list, conflict, and user-conflict flows use one validated time-window path and keep overlap analysis behind the event scheduling module interface.

## Type

- [ ] 🧪 Experiment (fitness function research)
- [ ] ✨ Feature (new business logic)
- [x] 🐛 Bug fix
- [ ] ♻️ Refactor / neutral change
- [ ] 📝 Documentation only

## Changes

- Added `CalendarTimeWindow` for validated half-open event windows.
- Added `EventConflictAnalyzer` for pairwise overlap detection and conflict DTO mapping.
- Refactored `EventServiceImpl` to reuse the shared time-window path.
- Fixed invalid partial event updates so validation happens before entity mutation.
- Bound conflict requester checks to `X-User-Id` to prevent spoofing `requesterId`.
- Added focused unit tests for window validation, conflict analysis, controller requester checks, and service regressions.

## Files changed

| File | Change |
|------|--------|
| `src/main/java/io/backend/lined/event/service/CalendarTimeWindow.java` | Added validated half-open time-window value object. |
| `src/main/java/io/backend/lined/event/service/EventConflictAnalyzer.java` | Added scheduling seam for overlap/conflict analysis. |
| `src/main/java/io/backend/lined/event/service/EventServiceImpl.java` | Reused validated windows across event flows and added own-calendar check. |
| `src/main/java/io/backend/lined/event/api/EventController.java` | Bound conflict requester identity to `X-User-Id`. |
| `src/test/java/io/backend/lined/event/**` | Added and updated focused tests for the new behavior. |

## Expected result

| Metric | Baseline (main) | Branch | Direction |
|--------|----------------|--------|-----------|
| checkstyle_violations |  |  | neutral |
| spotbugs_total |  |  | neutral |
| line_coverage |  |  | increase |
| critical_violations |  |  | neutral |
| code_smells |  |  | neutral |
| duplicated_lines_density |  |  | neutral |
| **F score** |  |  | neutral/increase |
| **SonarQube QG** |  |  | neutral |

## Checklist

- [x] ./gradlew check passes locally
- [ ] ./gradlew jacocoTestReport passes locally
- [x] No unintended changes to main business logic
- [x] Branch name matches experiment/feature naming convention

### CODE REVIEW

File: backend/lined/src/main/java/io/backend/lined/event/service/EventConflictAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                           List<EventConflictDto> conflicts) {
+    var firstWindow = windowOf(first);
+    var secondWindow = windowOf(second);
+    firstWindow.overlapWith(secondWindow).ifPresent(overlap ->
+        conflicts.add(new EventConflictDto(
+            mapper.toDto(first), mapper.toDto(second), overlap.start(), overlap.end())));
+  }
+
+  private CalendarTimeWindow windowOf(EventEntity event) {
+    return CalendarTimeWindow.of(event.getStartAt(), event.getEndAt(), ENTITY_WINDOW_MESSAGE);
```
Comment (Pan14ek) [inline backend/lined/src/main/java/io/backend/lined/event/service/EventConflictAnalyzer.java]: **[Bug risk]** `CalendarTimeWindow.of()` throws `BadRequestException` when called on a stored entity with invalid timestamps (e.g. `startAt == null` or `startAt >= endAt` from a migration defect or old data). That exception maps to HTTP 400, so the client receives a client-error response for what is a server-side data-integrity failure.

Consider detecting bad stored state separately and throwing `IllegalStateException` instead, letting the global handler map it to 500:
```java
private CalendarTimeWindow windowOf(EventEntity event) {
  if (event.getStartAt() == null || event.getEndAt() == null
      || !event.getStartAt().isBefore(event.getEndAt())) {
    throw new IllegalStateException(
        "Stored event %d has invalid time window".formatted(event.getId()));
  }
  return new CalendarTimeWindow(event.getStartAt(), event.getEndAt());
}
```

**Review by Pan14ek — COMMENTED** [review]
## Code Review Summary

Good refactor overall. The `CalendarTimeWindow` value object centralises validation cleanly, the `EventConflictAnalyzer` extraction removes duplication, the mutation-before-validation bug in `update()` is fixed, and the spoofable `requesterId` is now locked to the `X-User-Id` header.

Two things to address before merge:
1. **Missing regression test** for a partial update where one timestamp is `null` in the DTO but the resolved pair is invalid (see inline comment on `EventServiceImpl`).
2. **`BadRequestException` thrown from persisted entity data** in `EventConflictAnalyzer` (see inline comment) — maps a data-integrity failure to HTTP 400.

One post-merge candidate: the now-redundant `requesterId` query param on the conflict endpoints (see inline comment on the controller).

File: backend/lined/src/main/java/io/backend/lined/event/service/EventServiceImpl.java [hunk 2]
```diff
@@ -61,27 +59,22 @@ public EventDto create(EventCreateDto dto, Long currentUserId) {
   public EventDto update(Long id, EventUpdateDto dto, Long currentUserId) {
     var e = mustEvent(id);
     accessPolicy.ensureMember(e.getLobby(), currentUserId);
+    var window = eventWindow(
```
Comment (Pan14ek) [inline backend/lined/src/main/java/io/backend/lined/event/service/EventServiceImpl.java:62]: **[Missing test]** The window is resolved correctly here — `null` timestamps fall back to the stored values before validation — but there is no test covering the case where one timestamp is supplied in the DTO and the combined result is invalid.

Example: stored event is `10:00–11:00`; DTO sends `startAt=12:00, endAt=null`. The resolved pair is `12:00–11:00`, which should throw `BadRequestException` *and leave the entity unchanged*. Without a test, the regression protection for the mutation-before-validation fix is incomplete.

Suggested test name:
```java
void update_throwsBadRequest_whenNewStartAtExceedsExistingEndAt()
```

File: backend/lined/src/main/java/io/backend/lined/event/api/EventController.java [hunk 3]
```diff
@@ -94,19 +96,31 @@ public ResponseEntity<List<EventConflictDto>> findConflicts(
       @RequestParam Long lobbyId,
       @RequestParam OffsetDateTime start,
       @RequestParam OffsetDateTime end,
-      @RequestParam Long requesterId) {
+      @RequestParam Long requesterId,
```
Comment (Pan14ek) [inline backend/lined/src/main/java/io/backend/lined/event/api/EventController.java:99]: **[Post-merge candidate]** `requesterId` is now validated to equal `currentUserId` and then `currentUserId` is forwarded to the service, making the query param redundant: every call that passes the guard supplies the same value twice, and every call that doesn't is rejected with 403. The param no longer carries information.

This creates an awkward API contract for callers. Recommend deprecating `requesterId` from both conflict endpoints and removing it in the next cycle — the `X-User-Id` header alone is sufficient. The same applies to the `/user-conflict` endpoint.

File: backend/lined/src/main/java/io/backend/lined/event/service/CalendarTimeWindow.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+   * Calculates the shared bounds between this window and another window.
+   *
+   * @param other validated window to compare with
+   * @return overlap bounds, or empty when the windows do not overlap
+   */
+  Optional<CalendarTimeWindow> overlapWith(CalendarTimeWindow other) {
+    if (!overlaps(other)) {
+      return Optional.empty();
+    }
+    return Optional.of(new CalendarTimeWindow(max(start, other.start), min(end, other.end)));
```
Comment (Pan14ek) [inline backend/lined/src/main/java/io/backend/lined/event/service/CalendarTimeWindow.java:52]: **[Minor]** The overlap result is constructed directly with `new CalendarTimeWindow(...)`, bypassing the `of()` factory. This is safe today (the overlap of two valid windows is always a valid window), but if validation in `of()` ever expands, this second construction path will silently skip it.

A small comment explaining the invariant makes the intent explicit:
```java
// overlap of two valid windows is always a valid window — no need to re-validate
return Optional.of(new CalendarTimeWindow(max(start, other.start), min(end, other.end)));
```

File: backend/lined/src/test/java/io/backend/lined/event/service/EventConflictAnalyzerTest.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    when(mapper.toDto(inner)).thenReturn(dto(2L));
+
+    List<EventConflictDto> result = analyzer.findConflicts(List.of(outer, inner));
+
+    assertThat(result).hasSize(1);
+    assertThat(result.get(0).overlapStart()).isEqualTo(inner.getStartAt());
+    assertThat(result.get(0).overlapEnd()).isEqualTo(inner.getEndAt());
+  }
+
+  @Test
```
Comment (Pan14ek) [inline backend/lined/src/test/java/io/backend/lined/event/service/EventConflictAnalyzerTest.java:94]: **[Missing test]** There is no test for the case where a stored `EventEntity` has invalid timestamps (`startAt == null`, `endAt == null`, or `startAt >= endAt`). This is exactly the path `windowOf()` will hit if bad data reaches the analyzer. A test here would catch a regression if the validation or exception type changes:
```java
@Test
void findConflicts_throwsWhenEntityHasInvalidWindow() {
  var bad = event(1L, base.plusHours(1), base); // startAt after endAt
  var good = event(2L, base, base.plusHours(2));

  assertThatThrownBy(() -> analyzer.findConflicts(List.of(bad, good)))
      .isInstanceOf(BadRequestException.class); // or IllegalStateException after the fix above
}
```

File: backend/lined/src/main/java/io/backend/lined/event/service/EventConflictAnalyzer.java (hunk 1 above)
Comment (Pan14ek) [reply backend/lined/src/main/java/io/backend/lined/event/service/EventConflictAnalyzer.java]: Fixed

File: backend/lined/src/main/java/io/backend/lined/event/service/CalendarTimeWindow.java (hunk 4 above)
Comment (Pan14ek) [reply backend/lined/src/main/java/io/backend/lined/event/service/CalendarTimeWindow.java:52]: Fixed

File: backend/lined/src/test/java/io/backend/lined/event/service/EventConflictAnalyzerTest.java (hunk 5 above)
Comment (Pan14ek) [reply backend/lined/src/test/java/io/backend/lined/event/service/EventConflictAnalyzerTest.java:94]: Fixed

File: backend/lined/src/main/java/io/backend/lined/event/api/EventController.java (hunk 3 above)
Comment (Pan14ek) [reply backend/lined/src/main/java/io/backend/lined/event/api/EventController.java:99]: Fixed

File: backend/lined/src/main/java/io/backend/lined/event/service/EventServiceImpl.java (hunk 2 above)
Comment (Pan14ek) [reply backend/lined/src/main/java/io/backend/lined/event/service/EventServiceImpl.java:62]: Fixed

**Review by Pan14ek — COMMENTED** [review]
All review comments addressed — LGTM.

**What changed since the first review:**
- `EventConflictAnalyzer.windowOf()` now throws `IllegalStateException` for stored entities with invalid timestamps, correctly separating data-integrity failures from client errors.
- Windows are now pre-computed via `.stream().map(this::windowOf).toList()` before the O(n²) loop — a bonus improvement: each entity's window is built exactly once instead of potentially multiple times across `addConflict` calls.
- `CalendarTimeWindow.overlapWith()` has the invariant comment explaining why the direct constructor call is safe.
- `update_throwsBadRequest_whenNewStartAtExceedsExistingEndAt` covers the missing partial-update regression path (new `startAt` beyond existing `endAt` with a `null` DTO `endAt`).
- `EventConflictAnalyzerTest` adds two `IllegalStateException` tests for entities with invalid stored windows.

The redundant `requesterId` query param remains as a post-merge candidate — no objection.
